### ISSUE TITLE
[BUG] Terramate run becomes unresponsive when encountering errors

### ISSUE THREAD

**Issue by bgrams** [terramate-io/terramate#1828]
**Describe the bug**
`terramate run` will often become unresponsive when running parallel commands that encounter errors. This was observed both locally and in CICD.

The repro example contains a timeout however in practice terramate will hang indefinitely and not respond to interrupt signals.

**To Reproduce**
```bash
brandongr brandongr$ git init tm-bug
Initialized empty Git repository in /Users/brandongr/gitlab/brandongr/tm-bug/.git/
brandongr brandongr$ cd tm-bug/
tm-bug brandongr$ for i in $(seq 1 5); do terramate create stacks/$i; done
Created stack /stacks/1
Created stack /stacks/2
Created stack /stacks/3
Created stack /stacks/4
Created stack /stacks/5
tm-bug brandongr$ timeout 30 terramate run --parallel 2 -- cat foo.tf
terramate: Entering stack in /stacks/2
terramate: Entering stack in /stacks/5
terramate: Executing command "cat foo.tf"
terramate: Executing command "cat foo.tf"
cat: foo.tf: No such file or directory
terramate: Entering stack in /stacks/1
terramate: Executing command "cat foo.tf"
cat: foo.tf: No such file or directory
terramate: Entering stack in /stacks/4
cat: foo.tf: No such file or directory
terramate: Entering stack in /stacks/3
Error: one or more commands failed
> execution failed: running /bin/cat foo.tf (in /stacks/5): exit status 1
> execution failed: running /bin/cat foo.tf (in /stacks/2): exit status 1
> execution failed: running /bin/cat foo.tf (in /stacks/1): exit status 1
tm-bug brandongr$ for i in $(seq 6 10); do terramate create stacks/$i; done
Created stack /stacks/6
Created stack /stacks/7
Created stack /stacks/8
Created stack /stacks/9
Created stack /stacks/10
tm-bug brandongr$ timeout 30 terramate run --parallel 2 -- cat foo.tf
terramate: Entering stack in /stacks/2
terramate: Entering stack in /stacks/6
terramate: Executing command "cat foo.tf"
terramate: Executing command "cat foo.tf"
cat: foo.tf: No such file or directory
terramate: Entering stack in /stacks/7
cat: foo.tf: No such file or directory
terramate: Entering stack in /stacks/9
tm-bug brandongr$ echo $?
124
```

1. Init a git respository
2. Create stacks
3. Execute `terramate run --parallel n <some failing command>`, where num. stacks > n > 1

**Expected behavior**
Terramate execution predictably terminates with errors

**Log Output**
[tm-bug.log](https://github.com/user-attachments/files/16564090/tm-bug.log)

**Environment (please complete the following information):**
 - OS: Darwin
 - OS Version: 14.6
 - Git Version: 2.39.3 (Apple Git-146)
 - Terramate Version: 0.10.0

CICD environment is Docker 
Linux/Debian 12

**Additional context**
N/A

**Comment by cwe1ss** [comment]
We have the same issue with `script run` sometimes, but I haven't had time to diagnose this further yet.

**Comment by bgrams** [comment]
Working on a PR for this

**Comment by i4ki** [comment]
Hi @bgrams 
I was able to reproduce here and I will have a look at your PR now.
Coincidentally I ran into the same issue this week in our CI but I didn't have time to investigate.

Thanks for reporting and taking care of the fix!

**Comment by i4ki** [comment]
Closed by #1830 but more related fixes are on the way.
